packages/message/src/StepFunctions.ts

```diff
@@ -91,3 +112,133 @@ export const parseStepMessage = async (
   }
   return <Message.CumulusMessage>parsedMessage;
 };
+
+/**
+ * Searches the Execution step History for the TaskStateEntered pertaining to
+ * the failed task Id.  HistoryEvent ids are numbered sequentially, starting at
+ * one.
+*
+ * @param {HistoryEvent[]} events - Step Function events array
+ * @param {HistoryEvent} failedStepEvent - Step Function's failed event.
+ * @returns {string} name of the current stepfunction task or 'UnknownFailedStepName'.
+ */
+export const getFailedStepName = (
+  events: AWS.StepFunctions.HistoryEvent[],
+  failedStepEvent: { id: number }
+) => {
+  try {
+    const previousEvents = events.slice(0, failedStepEvent.id - 1);
+    const startEvents = previousEvents.filter(
+      (e) => e.type === 'TaskStateEntered'
+    );
+    return (
+      startEvents.pop()?.stateEnteredEventDetails?.name || 'UnknownFailedStepName'
```

Comment:
Is the `|| 'UnknownFailedStepName'` necessary given the default return on line 141?

Reply:
Sort of.   This method is intended to return `UnknownFailedStepName` in case of error *or* in case `events` takes an unexpected form and the filter fails/etc.      If you remove the conditional here it will return null successfully thanks to  converting to typescript/updating to optional chaining.   

That said, the logging will never be triggered, so this needs updated, and this was clearly just a 'fixing typescript error' oversight.    This 82bb41ec0 should be more equivalent logic. 

Nice catch.